lib/presentation/volunteer/accomodation/accomodation_list_page.dart

```diff
@@ -0,0 +1,189 @@
+            ),
+            loadFailure: (failureState) => Center(
+              child: Column(
+                children: [
+                  const Icon(
+                    Icons.error_outline,
+                    size: 72,
+                  ),
+                  Text(
+                    // TODO: failures should probably be mapped to specific user friendly messages for profuction
+                    failureState.failure.toString(),
+                  ),
+                ],
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class AccomodationListLoadSuccessWidget extends StatelessWidget {
+  const AccomodationListLoadSuccessWidget({
+    Key? key,
+    required this.accomodations,
+  }) : super(key: key);
+
+  final List<Accomodation> accomodations;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(4.0),
+      child: ListView(
+        children: [
+          const Padding(
+            padding: EdgeInsets.fromLTRB(0, 14, 0, 14),
+            child: Text(
+              'Lista zgłoszonych mieszkań',
+              textAlign: TextAlign.center,
+            ),
+          ),
+          ...accomodations
+              .map(
+                (accomodation) => Padding(
+                  padding: const EdgeInsets.fromLTRB(0, 4, 0, 4),
+                  child: ListTile(
+                    tileColor: () {
+                      switch (accomodation.status) {
+                        case AccomodationVerificationStatus.phoneVerified:
+                          return Colors.yellow[100];
+                        case AccomodationVerificationStatus.inPersonVerified:
+                          return Colors.green[200];
+                        case AccomodationVerificationStatus.added:
+                        default:
+                          return Colors.grey[200];
+                      }
+                    }(),
+                    leading: () {
+                      switch (accomodation.status) {
+                        case AccomodationVerificationStatus.phoneVerified:
+                          return Stack(
+                            children: const [
+                              Icon(Icons.phone),
+                              Icon(
+                                Icons.check,
+                                color: Colors.green,
+                                size: 35,
+                              ),
+                            ],
+                          );
+                        case AccomodationVerificationStatus.inPersonVerified:
+                          return Stack(
+                            children: const [
+                              Icon(Icons.person),
+                              Icon(
+                                Icons.check,
+                                color: Colors.green,
+                                size: 35,
+                              ),
+                            ],
```

Comment:
Za dużo tu zagnieżdżeń i przez to mało czytelne to jest.
Zrobiłbym tileColor i leading w osobnych funkcjach.

Reply:
Moved this to separate methods in a new commit